### ISSUE TITLE
Setting updateAndDelete to false will stop files from syncing

### ISSUE THREAD

**Issue by pitAlex** [mrmlnc/syncy#40]
Just like the title mention, when settings updateAndDelete to false files are no longer synced with the destination directory.

**Comment by mrmlnc** [comment]
You can give me more detailed description? Now i cannot reproduce your problem.

![qwe](https://cloud.githubusercontent.com/assets/7034281/24102651/b934d5ea-0d8d-11e7-95cb-4929638cea48.png)

**Comment by pitAlex** [comment]
So first run will work, its afterwards then it doesn't do anything. For my issue I am syncing minified js from one place to another. I first minify with gulp-uglify then I run syncy throught a gulp task, like this:

gulp.task('web-sync', function (done) {
        fsync([`${ROOT}/packs/minified/web-src/**/*.js`,`${ROOT}/packs/minified/web-src/*.js`], `${WEB_ROOT}/`, { verbose : true, updateAndDelete: false, base: `${ROOT}/packs/minified/web-src` })
        .then(() => {
            done();
        }).catch((e) => {
            console.log(e);
        });
});`

**Comment by mrmlnc** [comment]
Well, I also cannot reproduce your problem. Maybe you can provide demo repository?

Some questions:

  * OS?
  * Version of Node.js & npm?
  * Version of Gulp?
  * The `web-sync` task run only after `gulp-uglify`?

Also, please give me the output of this script:

```js
'use strict';

const fs = require('fs');
const syncy = require('syncy');

fs.mkdirSync('./fixtures');
fs.writeFileSync('./fixtures/a', 'a');
fs.writeFileSync('./fixtures/b', 'b');
fs.writeFileSync('./fixtures/c', 'c');

const files = ['./fixtures/**/*', './fixtures/*'];
const opts = {
	verbose: true,
	updateAndDelete: false,
	base: './fixtures'
};

console.log('One');
syncy(files, 'dest', opts)
	.then(() => fs.writeFileSync('./fixtures/d', 'd'))
	.then(() => {
		console.log('Two');
		return syncy(files, 'dest', opts)
	})
	.then(() => {
		console.log('Three');
		return syncy(files, 'dest', opts)
	});
```

**Comment by pitAlex** [comment]
I'm on mac sierra, and here is how it fails using the code above:

`
'use strict';

const fs = require('fs');
const syncy = require('syncy');

fs.mkdirSync('./fixtures');
fs.writeFileSync('./fixtures/a', 'a');
fs.writeFileSync('./fixtures/b', 'b');
fs.writeFileSync('./fixtures/c', 'c');

const files = ['./fixtures/**/*', './fixtures/*'];
const opts = {
	verbose: true,
	updateAndDelete: false,
	base: './fixtures'
};

console.log('One');
syncy(files, 'dest', opts)
	.then(() => fs.writeFileSync('./fixtures/d', 'd'))
	.then(() => {
		console.log('Two');
		return syncy(files, 'dest', opts)
	})
	.then(() => {
                fs.writeFileSync('./fixtures/a', 'd');
		console.log('Three');
		return syncy(files, 'dest', opts)
	});
`

The last part 'Three', does not sync 'a', who now has "d" as content, do "dest".

**Comment by mrmlnc** [comment]
This is correct behavior because you set `updateAndDelete` option as `false`. With this option Syncy will be only add new files (not update) to dest directory.

**Comment by pitAlex** [comment]
Thats what I suspected, however updateAndDelete means also delete any files that are in "dest" but not in "src", correct? You should make the delete option separately then, because the destination folder could be one that is synced from multiple places.

**Comment by brandondahler** [comment]
Appears to be related to the following code .  Specifically, it appears that setting updateAndDeleted to false tells it to skip updating files if they already exist in the destination.

https://github.com/mrmlnc/syncy/blob/master/src/syncy.ts#L24
```
export function skipUpdate(source: fs.Stats, dest: fs.Stats | null, updateAndDelete: boolean): boolean {
	if (dest && !updateAndDelete) {
		return true;
	}
```
